### PR TITLE
fix: Clamp VaultClawback to assetsAvailable for zero-amount clawback

### PR DESCRIPTION
**Pull request by Tapanito**
Zero-amount clawback (meaning "clawback all") returned early without clamping to assetsAvailable, allowing more assets to be recovered than available when there was an outstanding loan. Move the zero-amount path inside the try block so it shares the same clamping logic as non-zero clawback.

The old early-return behavior is retained behind an amendment gate for ledger replay compatibility.

### API Impact

- [ ] Public API: New feature (new methods and/or new fields)
- [ ] Public API: Breaking change (in general, breaking changes should only impact the next api_version)
- [ ] `libxrpl` change (any change that may affect `libxrpl` or dependents of `libxrpl`)
- [ ] Peer protocol change (must be backward compatible or bump the peer protocol version)

### CODE REVIEW

File: src/test/app/Vault_test.cpp [hunk 1]
```diff
@@ -4798,6 +4799,7 @@ class Vault_test : public beast::unit_test::suite
 
             auto const& vaultSle = env.le(vaultKeylet);
             BEAST_EXPECT(vaultSle != nullptr);
+            env.memoize(Account("vault", vaultSle->at(sfAccount)));
```
[the remark this answers is not in the document]
Comment (Tapanito) [reply src/test/app/Vault_test.cpp:4862]: Vault_test.cpp:4802`** — `setupVault` re-memoizes `"vault"` with a different account ID on each call, silently overwriting prior entries. Suggests using a unique name per call (e.g. including the keylet key).

This is about pre-existing code in `setupVault`, not something introduced by our changes. The memoize call was there before our commit. That said, it's a valid observation — but since each test case creates its own vault and doesn't look up the old one, the overwrite is harmless in practice

File: src/test/app/Vault_test.cpp [hunk 2]
```diff
@@ -4954,6 +4956,124 @@ class Vault_test : public beast::unit_test::suite
...
+            {
+                testcase(
+                    "VaultClawback (asset) - " + prefix +
+                    " zero-amount clawback clamped with outstanding loan");
+                auto [vault, vaultKeylet] = setupVault(asset, owner, depositor, issuer);
+
+                auto const vaultSle = env.le(vaultKeylet);
+                BEAST_EXPECT(vaultSle != nullptr);
+                if (!vaultSle)
+                    return;
```
Comment (shawnxie999) [inline src/test/app/Vault_test.cpp]: nit - alternatively could be (same for places down below)

```
if (!BEAST_EXPECT(vaultSle))
    return;
```

**Review by shawnxie999 — APPROVED** [review]
LGTM - left minor comment

File: src/test/app/Vault_test.cpp [hunk 3]
```diff
@@ -4954,6 +4956,124 @@ class Vault_test : public beast::unit_test::suite
...
+                env.close();
+
+                {
+                    auto const sle = env.le(vaultKeylet);
+                    BEAST_EXPECT(sle->at(sfAssetsAvailable) == asset(60).value());
+                    BEAST_EXPECT(sle->at(sfAssetsTotal) == asset(100).value());
+                }
+
+                auto const sharesBefore = env.balance(depositor, shares);
+
```
[the remark this answers is not in the document]
Comment (Tapanito) [reply src/test/app/Vault_test.cpp]: Resolved.

File: src/test/app/Vault_test.cpp (hunk 2 above)
Comment (Tapanito) [reply src/test/app/Vault_test.cpp]: Resolved.

File: src/test/app/Vault_test.cpp [hunk 4]
```diff
@@ -4988,6 +5101,74 @@ class Vault_test : public beast::unit_test::suite
...
+                // Transaction rolled back — vault and shares unchanged
+                auto const sle = env.le(vaultKeylet);
+                BEAST_EXPECT(sle != nullptr);
+                BEAST_EXPECT(sle->at(sfAssetsAvailable) == IOU(60).value());
+                BEAST_EXPECT(sle->at(sfAssetsTotal) == IOU(100).value());
+                auto const sharesAfter = env.balance(depositor, shares);
+                BEAST_EXPECT(sharesAfter == sharesBefore);
+            }
+
+            env.enableFeature(fixSecurity3_1_3);
```
Comment (pratikmankawde) [inline src/test/app/Vault_test.cpp:5399]: Neither tests covers the boundary conditions. Could be good to add this:
| Boundary case | Covered? | Notes
-- | -- | --
Partial clawback < available (e.g. clawback 30 when 60 available) | No | No test with loan + amount < assetsAvailable
Clawback == available (clawback exactly 60 when 60 available) | No | No test at the exact boundary
Zero assetsAvailable (100% borrowed) | No | No test with all assets loaned out
Non-1:1 ratio + loan | No | Scale tests exist but without loans; loan tests use 1:1 ratio

The existing "Scale clawback" tests (lines ~3908-4082) cover non-1:1 ratios but without any outstanding loans.

**Review by pratikmankawde — APPROVED** [review]
Looks good. Left minor comments on missing tests.

File: src/test/app/Vault_test.cpp (hunk 4 above)
Comment (Tapanito) [reply src/test/app/Vault_test.cpp:5399]: Added extra unit tests.

**Comment by mvadari** [comment]
@Tapanito how does this handle `sfLockedAmount` in MPTs?

**Comment by Tapanito** [comment]
@mvadari good question, I added vault unit tests to ensure that `sfLockedAmount` is handled correctly.
